Regular Expressions/syntax_test_regexp.re

```diff
@@ -324,3 +324,83 @@ hello++
 #    ^^ keyword.control.character-class.regexp
 #      ^ keyword.operator.quantifier.regexp
 #       ^^ constant.character.escape.regexp - keyword.control.set.regexp
+
+(?<test>a)?b(?('test')c|d)
+#            ^^^^^^^^^ keyword.other.backref-and-recursion.conditional.regexp
+(?<test>a)?b(?(<test>)c|d)
+#            ^^^^^^^^^ keyword.other.backref-and-recursion.conditional.regexp
+(a)?b(?(1)c|d)
+#     ^^^^ keyword.other.backref-and-recursion.conditional.regexp
+(?(?!\d)a|b)
+#^ keyword.other.conditional.regexp
+# ^ keyword.control.group.regexp
+#  ^^ constant.other.assertion.regexp
+#    ^^ keyword.control.character-class.regexp
+#      ^ keyword.control.group.regexp
+#       ^ meta.literal.regexp
+#        ^ keyword.operator.regexp
+#         ^ meta.literal.regexp
+#          ^ keyword.control.group.regexp
+(?(?=\d)\da|b)
+#^ keyword.other.conditional.regexp
+# ^ keyword.control.group.regexp
+#  ^^ constant.other.assertion.regexp
+#    ^^ keyword.control.character-class.regexp
+#      ^ keyword.control.group.regexp
+#       ^^ keyword.control.character-class.regexp
+#         ^ meta.literal.regexp
+#          ^ keyword.operator.regexp
+#           ^ meta.literal.regexp
+#            ^ keyword.control.group.regexp
+(?(?<=\dc)a|b)
```

Comment:
it is, it's a lookbehind :) unless I'm blind and have made a typo I can't spot atm :) http://www.boost.org/doc/libs/1_44_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.lookbehind